mycroft/enclosure/gui.py

```diff
@@ -339,6 +381,54 @@ def show_url(self, url, override_idle=None,
         self.show_page("SYSTEM_UrlFrame.qml", override_idle,
                        override_animations)
 
+    def play_video(self, url, title="", override_idle=True,
+                   override_animations=None):
+        """ Play video stream """
+        self["playStatus"] = "play"
+        self["video"] = url
+        self["title"] = title
+        self.video_info = {"title": title, "url": url}
+        self.show_page("SYSTEM_VideoPlayer.qml",
+                       override_idle=override_idle,
+                       override_animations=override_animations)
+
+    @property
+    def is_video_displayed(self):
+        """Returns whether the gui is in a video playback state.
+        Eg if the video is paused, it would still be displayed on screen
+        but the video itself is not "playing" so to speak"""
+        return self.video_info is not None
+
+    @property
+    def playback_status(self):
+        """Returns gui playback status,
+        indicates if gui is playing, paused or stopped"""
+        if self.__session_data.get("playStatus", -1) == "play":
+            return GUIPlaybackStatus.PLAYING
+        if self.__session_data.get("playStatus", -1) == "pause":
+            return GUIPlaybackStatus.PAUSED
+        if self.__session_data.get("playStatus", -1) == "stop":
+            return GUIPlaybackStatus.STOPPED
+        return GUIPlaybackStatus.UNDEFINED
+
+    def pause_video(self):
+        """Pause video playback."""
+        if self.is_video_displayed:
+            self["playStatus"] = "pause"
+
+    def stop_video(self):
+        """Stop video playback."""
+        # TODO detect end of media playback from gui and call this
```

Comment:
@AIIX do you have any suggestions for this TODO ?

Reply:
yes there is, we can detect when the media has ended or stopped playing, or status has changed and call the above https://doc.qt.io/qt-5/qml-qtmultimedia-video.html#status-prop